packages/medusa/src/services/cart.ts

```diff
@@ -831,6 +831,12 @@ class CartService extends TransactionBaseService {
           cart.customer = customer
           cart.customer_id = customer.id
           cart.email = customer.email
+        } else if (data.customer_id === "") {
+          type Nullable<T> = { [K in keyof T]: T[K] | null };
```

Comment:
suggestion: If the property on the entity is nullable, it should be that property that accepts `| null`. The problem here is that if we continue that way, it means that the other places using the cart have no idea that the customer can be null or undefined and have to guess it and therefore the type checking will not warn about it which can lead to more bugs. wdyt?

It would mean:
- updating the cart entity properties types such as `customer?: Customer | null`
- removing this local type

Reply:
Done